### ISSUE TITLE
Drop in accuracy after saving

### ISSUE THREAD

**Issue by arfangeta** [microsoft/hummingbird#218]
I wanted to ask, what could be the reason for the drop in accuracy after saving in this sample? 

```from sklearn.ensemble import RandomForestClassifier
from sklearn.datasets import load_breast_cancer
from onnxmltools import convert_sklearn
from onnxmltools.convert.common.data_types import FloatTensorType
from hummingbird.ml import convert
from sklearn.model_selection import train_test_split
import numpy as np
from sklearn.metrics import accuracy_score
import onnxruntime as ort

X, y = load_breast_cancer(return_X_y=True)
X = X.astype('|f4')
X, X_test, y, y_test = train_test_split(X, y, test_size=0.5, shuffle=True)


skl_model = RandomForestClassifier(n_estimators=500, max_depth=7)
skl_model.fit(X, y)

predict = skl_model.predict(X_test)
print('accuracy', accuracy_score(y_test, predict))


initial_type = [('float_input', FloatTensorType([X_test.shape[0], X_test.shape[1]]))]
onnx_ml_model = convert_sklearn(skl_model, initial_types=initial_type)

onnx_model = convert(onnx_ml_model, "onnx", X_test)
with open("random_forest.onnx", "wb") as f:
    f.write(onnx_model.SerializeToString())

np.save('x_test.npy', X_test)
np.save ('y_test.npy', y_test)


sess = ort.InferenceSession("random_forest.onnx")
input_name = sess.get_inputs()[0].name
label_name = sess.get_outputs()[0].name


pred_onx = sess.run([label_name], {input_name: X_test})[0]
print('accuracy after', accuracy_score(y_test, pred_onx))


_______________________
accuracy 0.9473684210526315
accuracy after 0.7052631578947368

**Comment by arfangeta** [comment]
And a similar question, but without saving, I get a small drop. Is this my mistake or should it be?
```import torch
import numpy as np

import onnxruntime as ort
from onnxmltools import convert_sklearn
from onnxconverter_common.data_types import FloatTensorType

from hummingbird.ml import convert
from hummingbird.ml import constants

from sklearn.metrics import accuracy_score
from sklearn.ensemble import RandomForestClassifier

num_classes = 2
X = np.array(np.random.rand(10000, 28), dtype=np.float32)
y = np.random.randint(num_classes, size=10000)
X_test = np.array(np.random.rand(10000, 28), dtype=np.float32)
y_test = np.random.randint(num_classes, size=10000)

model = RandomForestClassifier(n_estimators=10, max_depth=10)
model.fit(X, y)
predict = model.predict(X_test)
print('accuracy', accuracy_score(y_test, predict))


initial_types = [("input", FloatTensorType([X.shape[0], X.shape[1]]))] # Define the inputs for the ONNX
onnx_ml_model = convert_sklearn(
    model, initial_types=initial_types, target_opset=9
)

onnx_model = convert(onnx_ml_model, "onnx", X)

session = ort.InferenceSession(onnx_model.SerializeToString())
output_names = [session.get_outputs()[i].name for i in range(len(session.get_outputs()))]
inputs = {session.get_inputs()[0].name: X_test}

predict_2 = session.run(output_names, inputs)[0]
print('accuracy after', accuracy_score(y_test, predict_2))
________
accuracy 0.5061
accuracy after 0.4924

**Comment by interesaaat** [comment]
Hi @arfangeta, thanks for reaching out! What you are hitting is not a problem with saving but there is actually a bug on the decision tree / random forest converter for onnx. I already fixed regression in a branch. I am looking right now into classification. Hopefully I will have a PR by early next week.

**Comment by interesaaat** [comment]
PR #222 should fix this. @arfangeta can you please try once it is merged?

**Comment by arfangeta** [comment]
thank you, it works :)
